### PR TITLE
Allow float() to be called without an argument

### LINKED ISSUE #1421: No argument with float function do not return 0.0
# Feature
``` python3
print(float())
```

If no argument is given, float function have to return 0.0

# Actual result
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Expected at least 2 arguments (1 given)
```

# Expected result
```
0.0
```

### PR DESCRIPTION
**Pull request by dralley**
Copy the same pattern used by int()

closes #1421

### CODE REVIEW

File: vm/src/obj/objfloat.rs [hunk 1]
```diff
@@ -667,4 +686,7 @@ impl TryFromObject for IntoPyFloat {
 #[rustfmt::skip] // to avoid line splitting
 pub fn init(context: &PyContext) {
     PyFloat::extend_class(context, &context.types.float_type);
+    extend_class!(context, &context.types.float_type, {
+        "__new__" => context.new_rustfunc(float_new),
+    });
```
Comment (coolreader18) [inline vm/src/obj/objfloat.rs]: You should keep `float_new` in the `pyimpl`, it's a really simple function anyway.

File: vm/src/obj/objfloat.rs [hunk 2]
```diff
@@ -574,6 +540,59 @@ impl PyFloat {
     }
 }
 
+#[derive(FromArgs)]
+struct FloatOptions {
+    #[pyarg(positional_only, optional = true)]
+    val_options: OptionalArg<PyObjectRef>,
+}
```
Comment (coolreader18) [inline vm/src/obj/objfloat.rs]: I don't think this separate options struct is necessary, it's positional by default when you just use `OptionalArg` as a function argument.

File: vm/src/obj/objfloat.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/obj/objfloat.rs]: I do like the separation of `to_float` out into its own function though, nice work!

**Comment by dralley** [comment]
@coolreader18 When I try this, it panics. Any idea what's going on?

```
>>>>> float()
thread 'main' panicked at 'unexpected payload for type', src/libcore/option.rs:1190:5
```

Commit pushed so you can see the changes.

File: vm/src/obj/objfloat.rs [hunk 3]
```diff
@@ -342,6 +342,16 @@ impl PyFloat {
         -self.value
     }
 
+    #[pymethod(name = "__pow__")]
```
Comment (coolreader18) [inline vm/src/obj/objfloat.rs]: ^^^^

Should be `__new__`

Comment (dralley) [reply vm/src/obj/objfloat.rs]: I feel really, really dumb.  Thanks.

File: vm/src/obj/objfloat.rs (hunk 2 above)
Comment (dralley) [reply vm/src/obj/objfloat.rs]: I actually had no idea you could do that, neat!

File: vm/src/obj/objfloat.rs (hunk 1 above)
Comment (dralley) [reply vm/src/obj/objfloat.rs]: I actually had no idea you could do that, neat!

**Comment by coolreader18** [comment]
Looks like the rustfmt job is failing.

**Pull request merged**